internal/hud/webview/convert.go

```diff
@@ -295,12 +295,55 @@ func toUIResource(mt *store.ManifestTarget, s store.EngineState, disableSources
 	if err != nil {
 		return nil, err
 	}
+
+	r.Status.Conditions = []v1alpha1.UIResourceCondition{
+		UIResourceReadyCondition(r.Status),
+	}
 	return r, nil
 }
 
+// The "Ready" condition is a cross-resource status report that's synthesized
+// from the more type-specific fields of UIResource.
+func UIResourceReadyCondition(r v1alpha1.UIResourceStatus) v1alpha1.UIResourceCondition {
+	c := v1alpha1.UIResourceCondition{
+		Type:   v1alpha1.UIResourceReady,
+		Status: metav1.ConditionUnknown,
+
+		// LastTransitionTime will be computed by diffing against the current
+		// Condition. This doesn't really fit into the usual reconciler pattern,
+		// but is considered a worthwhile trade-off for the semantics we want, see discussion here:
+		// https://maelvls.dev/kubernetes-conditions/
+		LastTransitionTime: apis.NowMicro(),
+	}
+
+	if r.RuntimeStatus == v1alpha1.RuntimeStatusOK {
+		c.Status = metav1.ConditionTrue
+		return c
+	}
+
+	if r.RuntimeStatus == v1alpha1.RuntimeStatusNotApplicable && r.UpdateStatus == v1alpha1.UpdateStatusOK {
+		c.Status = metav1.ConditionTrue
+		return c
```

Comment:
The reasons in the failure cases are nifty - I wonder if we should have them in the success cases as well